spec/ndx-pose.extensions.yaml

```diff
@@ -91,27 +120,86 @@ groups:
       dtype: text
       doc: Version string of the software tool used.
       required: false
-  - name: nodes
+  groups:
+  - neurodata_type_inc: PoseEstimationSeries
+    doc: Estimated position data for each body part.
+    quantity: '*'
+  links:
+  - target_type: Skeleton
+    doc: Layout of body part locations and connections.
+- neurodata_type_def: TrainingFrame
+  neurodata_type_inc: NWBDataInterface
+  default_name: TrainingFrame
+  doc: Group that holds ground-truth position data for all instances in a single frame.
+  attributes:
+  - name: annotator
     dtype: text
-    dims:
-    - num_body_parts
-    shape:
-    - null
-    doc: Array of body part names corresponding to the names of the PoseEstimationSeries
-      objects within this group.
+    doc: Name of annotator who labeled the TrainingFrame.
+    required: false
+  groups:
+  - neurodata_type_inc: Instance
+    doc: Position data for a single instance in a single training frame.
+    quantity: '*'
+  - name: source_video
+    doc: Path to original video file and frame used.
     quantity: '?'
-  - name: edges
+    attributes:
+    - name: path
+      dtype: text
+      doc: Path to original video file.
+      required: false
```

Comment:
Should this and frame_index be optional? If a source_video is provided, it seems like a path and frame index should be provided.

Reply:
The logic of requirement I think was to be applied to the entirety of the `source_video` attribute. Combined logic being the user either links `source_video` to an `ImageSeries` (internal or external storage) plus the frame index used and then they also specify the `source_frame` as a standalone `Image` (ideally internal or external, but see other comments regarding that). Hence `source_frame` is required, but the video it was extracted from is optional (but ideal) provenance.

So should the `required` logic apply to `source_video` directly then?

Reply:
We wanted to provide two ways for supplying the training frame - either as a frame index given the path to the source video (`source_video`) or as a standalone image `source_frame`. Hence why neither are required although at least one must be supplied.

Reply:
I think we are confusing two different things here. I get that `source_video` is optional. If `source_video` is provided, then because the `path` and `frame_index` fields are optional, then the user does not have to provide either. From what you have both written, I *think* if `source_video` is provided, then `path` and `frame_index` should also be required, and we can specify that by removing `required: false` from both attributes.